app/_src/mesh/guides/create-a-mesh-using-terraform-and-konnect.md

```diff
@@ -0,0 +1,602 @@
+
+## Adding an example policy
+
+Policies examples now contain an additional tab called "Terraform" showing a terraform representation of a policy.
+
+Let's take an example from [MeshTrafficPermission page](/mesh/{{page.release}}/policies/meshtrafficpermission/#allow-all)
+
+We need to adjust it to our setup by adding the provider, the control plane id, mesh name, depends_on and labels:
+
+```hcl
+provider = konnect-beta
+cp_id    = konnect_mesh_control_plane.my_meshcontrolplane.id
+mesh     = konnect_mesh.mesh1.name
+depends_on = [konnect_mesh.mesh1]
+labels = {
+    "kuma.io/mesh" = konnect_mesh.mesh1.name
+}
+```
+
+{% warning %}
+Autogenerated labels like "kuma.io/mesh", "kuma.io/origin" etc. have to be manually added to the resources.
+This limitation will be removed in the GA release.
+{% endwarning %}
+
+The resulting resource looks like this:
+
+```hcl
+resource "konnect_mesh_traffic_permission" "allow_all" {
+    provider = konnect-beta
+    cp_id    = konnect_mesh_control_plane.my_meshcontrolplane.id
+    mesh     = konnect_mesh.mesh1.name
+    depends_on = [konnect_mesh.mesh1]
+    labels = {
+        "kuma.io/mesh" = konnect_mesh.mesh1.name
+    }
+    type = "MeshTrafficPermission"
+    name = "allow-all"
+    spec = {
+        from = [
+            {
+                target_ref = {
+                    kind = "Mesh"
+                }
+                default = {
+                    action = "Allow"
+                }
+            }
+        ]
+    }
+}
+```
+
+## Impact of Renaming Resources
+
+Certain properties (like Mesh name, policy name, etc.) are used as identifiers and changing them will result in a new resource being created and all dependant resources being recreated.
+
+So changing mesh name to `another-name`
+
+```hcl
+resource "konnect_mesh" "mesh1" {
+  # ...
+  name = "another-name"
+  # ...
+}
+```
+
+Will result in forced replacement of both `mesh` and `mesh_access_log` resources:
+
+```
+    # konnect_mesh.mesh1 must be replaced
+-/+ resource "konnect_mesh" "mesh1" {
+      ~ name                           = "mesh1" -> "another-name" # forces replacement
+        # (4 unchanged attributes hidden)
+    }
+
+  # konnect_mesh_access_log.konnect_mesh_traffic_permission must be replaced
+-/+ resource "konnect_mesh_traffic_permission" "allow_all" {
+      ~ creation_time     = "2025-03-13T09:53:00.606442Z" -> (known after apply)
+      ~ labels            = {
+          ~ "kuma.io/mesh" = "mesh1" -> "another-name"
+        }
+      ~ mesh              = "mesh1" -> "another-name" # forces replacement
+```
+
+## Deploying a Kubernetes Zone
+
+To deploy a Kubernetes zone you can use any Kubernetes services, in this guide we will use `k3d` to create a local Kubernetes cluster.
+
+Create a new k3d cluster:
+```bash
+k3d cluster create tfmink
+```
+
+Write `tfmink` cluster configuration:
+```bash
+export KUBECONFIG=$(k3d kubeconfig write tfmink)
```

Comment:
do you use the fact `KUBECONFIG` env is exported somewhere in TF? if no, maybe it should be just
```suggestion
k3d kubeconfig write tfmink
```

Reply:
only so that a user can then just https://github.com/Kong/docs.konghq.com/pull/8553/files#diff-b1938945f5af4d525de6474fa06dc2fd4187a5fe97ef535b9c62fb7d9c18ca7aR517 - do you think I should change it?